Sources_v3/WebSocketClient/Engine/StarscreamWebSocketEngine.swift

```diff
@@ -46,19 +46,28 @@ import Foundation
     }
     
     extension StarscreamWebSocketProvider: Starscream.WebSocketDelegate {
-        func websocketDidConnect(socket: Starscream.WebSocketClient) {
-            delegate?.webSocketDidConnect()
+        func didReceive(event: WebSocketEvent, client: WebSocket) {
+            switch event {
+            case .connected:
+                delegate?.webSocketDidConnect()
+            case let .text(text):
+                delegate?.webSocketDidReceiveMessage(text)
+            case let .error(error):
+                delegate?.webSocketDidDisconnect(error: error.map(WebSocketEngineError.init))
+            case let .disconnected(reason, code):
+                let error = WebSocketEngineError(reason: reason, code: Int(code), engineError: nil)
+                delegate?.webSocketDidDisconnect(error: error)
+            case .cancelled:
+                let error = WebSocketEngineError(
+                    reason: "Cancelled",
+                    code: WebSocketEngineError.stopErrorCode,
+                    engineError: nil
+                )
+                delegate?.webSocketDidDisconnect(error: error)
+            default:
```

Comment:
Here is the enum so you can see which cases trap into `default`:
```swift
public enum WebSocketEvent {
    case connected([String: String])
    case disconnected(String, UInt16)
    case text(String)
    case binary(Data)
    case pong(Data?)
    case ping(Data?)
    case error(Error?)
    case viabilityChanged(Bool)
    case reconnectSuggested(Bool)
    case cancelled
}
```
There is a `.reconnectSuggested` event, maybe we want to use it somehow instead/together with `WebSocketClientReconnectionStrategy`?

Reply:
Interesting. Did you test for example the `pong` case? Is it fired with the `text` case, too? In other words - is `WebSocketClient` still receiving pong events even via `text`?

Reply:
Checked it. The `.pong` event doesn't come with `.text`

Reply:
What we treat as **pong** is a `HealthCheckEvent` event and we receive it with `.text` event. So the `WebSocketClient` does receive **pong** events.

Do you think we should introduce `WebSocketEngineDelegate.webSocketDidReceivePong()`, invoke it when `.pong` event comes and create `HealthCheckEvent` manually with existed connection id?

Reply:
OK good, I understood it differently from your message. The important thing is `WebSocketClient` receives the health check events via `.text` 👍